### PR TITLE
Refactor category featured content API endpoint

### PR DESCRIPTION
**Pull request by davidthomas22**
refactor variable names in API Class and content resource

### CODE REVIEW

File: modules/custom/moj_resources/src/CategoryFeaturedContentApiClass.php [hunk 1]
```diff
@@ -19,7 +19,7 @@ class CategoryFeaturedContentApiClass
    *
    * @var array
    */
-  protected $nids = array();
+  protected $nodeIds = array();
```
Comment (psoleckimoj) [inline modules/custom/moj_resources/src/CategoryFeaturedContentApiClass.php:22]: This is not actually used. A variable with the same name is used locally in one of the functions but not this private one.

File: modules/custom/moj_resources/src/CategoryFeaturedContentApiClass.php [hunk 2]
```diff
@@ -83,53 +83,53 @@ protected function translateNode(NodeInterface $node)
...
   /**
-   * Get nids
+   * Get node ids
    *
    * @return void
    */
   protected function getFeaturedContentNodeIds($category, $number, $prison = 0)
   {
     $series = $this->promotedSeries($category, $prison);
     $nodes = $this->promotedNodes($category, $number, $prison);
```
Comment (psoleckimoj) [inline modules/custom/moj_resources/src/CategoryFeaturedContentApiClass.php:93]: In this context it'd make more sense to rename node to content as that's what it's pulling back.

File: modules/custom/moj_resources/src/CategoryFeaturedContentApiClass.php [hunk 3]
```diff
@@ -83,53 +83,53 @@ protected function translateNode(NodeInterface $node)
...
-    $results = array_merge($series, $nodes);
+    $query = array_merge($series, $nodes);
```
Comment (psoleckimoj) [inline modules/custom/moj_resources/src/CategoryFeaturedContentApiClass.php:94]: ...and this actually is results and not a Drupal query. The queries are within the `promoted<blah>` functions. Can go through it with you tomorrow if you like?

File: modules/custom/moj_resources/src/CategoryFeaturedContentApiClass.php [hunk 4]
```diff
@@ -208,28 +238,30 @@ private function promotedTerms($termIds, $prison)
...
    */
-  protected function serialize($item)
+  protected function serialize($term)
   {
-    $serializer = \Drupal::service($item->getType() . '.serializer.default'); // TODO: Inject dependency
-    return $serializer->serialize($item, 'json', ['plugin_id' => 'entity']);
+    $serializer = \Drupal::service($term->getType() . '.serializer.default'); // TODO: Inject dependency
+    return $serializer->serialize($term, 'json', ['plugin_id' => 'entity']);
   }
 }
```
Comment (LucasCairns) [inline modules/custom/moj_resources/src/CategoryFeaturedContentApiClass.php:267]: Do we use this method, if not we could remove?

**Review by psoleckimoj — APPROVED** [review]
LGTM

**Review by LucasCairns — APPROVED** [review]
LGTM 👍

**Pull request merged**